File: app/src/main/java/co/appdev/boilerplate/injection/module/ApplicationModule.java
package co.appdev.boilerplate.injection.module;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import co.appdev.boilerplate.injection.ApplicationContext;
import dagger.Module;
import dagger.Provides;

/**
 * Provide application-level dependencies.
 */
@Module
public class ApplicationModule {
    private final Application mApplication;

    public ApplicationModule(Application application) {
        mApplication = application;
    }

    @Provides
    Application provideApplication() {
        return mApplication;
    }

    @Provides
    @ApplicationContext
    Context provideContext() {
        return mApplication;
    }

}

File: app/src/main/java/co/appdev/boilerplate/di/module/ApplicationModule.kt
package co.appdev.boilerplate.di.module

import android.app.Application
import android.content.Context


import co.appdev.boilerplate.di.ApplicationContext

import dagger.Module
import dagger.Provides

@Module
class ApplicationModule(private val mApplication: Application) {

    @Provides
    internal fun provideApplication(): Application {
        return mApplication
    }

    @Provides
    @ApplicationContext
    internal fun provideContext(): Context {
        return mApplication
    }
}
